File: javascript/Q228/Q228.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */

var summaryRanges = function(nums) {
    let result = []

    if (nums.length === 0) {
        return []
    }
    
    let start = 0
    let end = 0

    while(true) {
        if (nums[end+1] === (nums[end] + 1)) {
            end = end + 1
        } else { 
            if (start === end) {
                result.push(""+nums[start])
                start = end + 1
                end = start
            } else {
                result.push(nums[start] + "->" + nums[end])
                start = end + 1 
                end = start
            }
        }
        
        if (nums[end] == null) {
            break
        }
    }

    return result
};

File: ruby/Q228/Q228.rb
# @param {Integer[]} nums
# @return {String[]}
def summary_ranges(nums)
  return [] if nums.length == 0

  result = []

  start = 0
  last = 0

  while true 
    while nums[last + 1] == nums[last] + 1 
      last = last + 1
      break if last == nums.length
    end

    if start == last
      ele = nums[last].to_s
    elsif last > start 
      ele = nums[start].to_s + "->" + nums[last].to_s
    end

    result.push(ele)
    start = last + 1
    last = start

    return result if start == nums.length
  end

end

